src/kakarot/instructions/environmental_information.cairo

```diff
@@ -79,6 +80,33 @@ namespace EnvironmentalInformation {
         return ctx;
     }
 
+    // @notice GASPRICE operation.
+    // @dev Get the size of return data.
+    // @custom:since Frontier
+    // @custom:group Environmental Information
+    // @custom:gas 2
+    // @custom:stack_consumed_elements 0
+    // @custom:stack_produced_elements 1
+    // @return The pointer to the updated execution context.
+    func exec_returngasprice{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
+        %{ print("0x3a - GASPRICE") %}
+        // Get caller address.
+        let return_gas_price = Helpers.to_uint256(ctx.intrinsic_gas_cost);
```

Comment:
Gas price is not the intrinsic gas cost

Reply:
Thank you for the feedback on this. What is the correct way to calculate gas price?

Reply:
We haven’t decided yet how to represent it. On Ethereum it was decided by the user (before EIP-1559). It will depend on how we represent EVM gas on our ZK EVM. I suggest we pause this PR until we take a decision. I will update you next week about this. Meanwhile feel free to work on another opcode or another issue

Reply:
Got it, makes sense. Thanks.